crates/node/types/src/block.rs

```diff
@@ -0,0 +1,113 @@
+//! Block abstraction.
+
+use alloc::fmt;
+use core::mem;
+
+use alloy_consensus::{BlockHeader, Requests, Transaction, TxType};
+use alloy_primitives::{Address, B256};
+use reth_primitives::{proofs, Withdrawals};
+
+/// Abstraction of block data type.
+pub trait Block: From<(Self::Header, Self::Body)> {
+    /// Header part of the block.
+    type Header: BlockHeader;
+    /// The block's body contains the transactions in the block.
+    type Body: BlockBody;
+}
+
+/// Abstraction for block's body.
+pub trait BlockBody:
+    Clone
+    + fmt::Debug
+    + PartialEq
+    + Eq
+    + Default
+    + serde::Serialize
+    + for<'de> serde::Deserialize<'de>
+    + alloy_rlp::Encodable
+    + alloy_rlp::Decodable
+{
+    /// Ordered list of signed transactions as committed in block.
+    // todo: requires trait for signed transaction
+    type SignedTransaction: Transaction;
+
+    /// Header type (uncle blocks).
+    type Header: BlockHeader;
+
+    /// Returns reference to transactions in block.
+    fn transactions(&self) -> &Vec<Self::SignedTransaction>;
+
+    /// Returns [`Withdrawals`] in the block, if any.
+    fn withdrawals(&self) -> Option<&Withdrawals>;
```

Comment:
this doesn't really belong in the core block trait because this bakes in withdrawals into the block, which may not be the case, same for ommers and requests.

but we can keep that for now because we only support these type of blocks atm